### DISCUSSION TITLE
Small Intelligent Hub (part 67351) from #45345 Spike Essential set.

### DISCUSSION THREAD

**Discussion by tomalphin** [pybricks/support#643] [Q&A]
Hey team,
I've asked a few questions in the past, especially when I was working on the first version of my general LEGO Powered Up guide (at https://brickarchitect.com/powered-up/ )

I'm working on a review of the Education-focused #45345 Spike Essential set, which includes a smaller hub that is similar to the 6-port hub included with Spike Prime or the new Mindstorms set.

QUESTION: Is it supported by PyBricks?  I have not seen any mention of this hub in the documentation, but maybe it isn't included in the documentation because it is such a rare part, not because it isn't supported.

I want to accurately reflect support in PyBricks app for both the 3×3 light matrix (which is supported by PyBricks) and the Hub.

Sincerely,
---tom

**Comment by laurensvalk** [comment]
Hi Tom, thanks for reaching out.

The SPIKE Essential Hub will be supported indeed. You can see a demo [here](https://www.youtube.com/watch?v=lTucJIUhj3k).

Just like on the SPIKE Prime Hub, saving programs and restarting them with the button will be possible.

Almost everything is already working. The only thing missing is an easy installation procedure in our app.

We're still working on that part, so that's why we haven't talked much about it yet :smile:

**Comment by tomalphin** [comment]
great, thanks!  Unless I hear otherwise, I will refer to support as "coming soon" instead of "not supported".

---tom

**Comment by tomalphin** [comment]
P.S. based on your analysis of the smaller hub, is it correct to say that it is functionally identical to the larger hub, except for fewer ports, no integrated 5x5 display, no speaker, and smaller battery?

(I suspect that the CPU/Memory would be identical.)

**Reply by dlech** [reply]
That sounds right. You can find the technical specifications at https://education.lego.com/en-us/product-resources/45345-spike-essential-resource-page/downloads/technical-specifications

**Reply by laurensvalk** [reply]
CPU and memory are the same, but it has less storage space for user files.

The SPIKE Prime Hub has 31 MB.

The SPIKE Essential Hub has 3MB. I'm not sure the spec sheet is correct on this one. This is still plenty, though. Programs don't take up much space.

By comparison, the other hubs have no dedicated storage for user programs at all.

**Comment by tomalphin** [comment]
3mb is an error.  It looks like this has been fixed on the LEGO website...

![image](https://user-images.githubusercontent.com/66934882/163587090-c38e6af9-5392-4760-ae63-251204c7417c.png)

**Reply by laurensvalk** [reply]
Right, I meant that the spec sheet might be incorrect. I believe it has 4MB storage, 1MB of which is reserved for system usage.

Similarly, SPIKE Prime has 32 - 1 = 31 MB for programs and content.

**Reply by tomalphin** [reply]
understood, thx.

**Reply by laurensvalk** [reply]
It wouldn't be that surprising when looking at what you need the space for. The only "large" files you would practically need are sound files, but the SPIKE Essential Hub doesn't have a speaker. It also doesn't have 19 code slots like SPIKE Prime.
